lib/prelude.dx

```diff
@@ -775,8 +775,8 @@ def dot [VSpace v] (s:n=>Float) (vs:n=>v) : v = sum for j. s.j .* vs.j
 def inner (x:n=>Float) (mat:n=>m=>Float) (y:m=>Float) : Float =
   fsum view (i,j). x.i * mat.i.j * y.j
 
-def eye [Eq n] : n=>n=>Float =
-  for i j. select (i == j) 1.0 0.0
+def eye [Eq n, Add a, Mul a] : n=>n=>a =
```

Comment:
We should really get a `Field` type class 😛

Reply:
Or maybe `Semiring` IIUC (as a field would have both additive and multiplicative inverses too).

Reply:
Yeah I guess `Semiring` is sufficient for many matrix multiplication, and it should be sufficient to produce an identity for it.